src/session/session-utils.c

```diff
@@ -64,6 +64,11 @@ read_authorize_response (const char *what)
   len = cockpit_frame_read (STDIN_FILENO, &message);
   if (len < 0)
     err (EX, "couldn't read %s", what);
+  if (len == 0)
+    {
+       debug ("EOF reading %s authorize message", what);
+       exit (0);
```

Comment:
The `err` and `errx` calls feel like `abort()` or `throw` type things, but this feels like an expected case, but it has this extraordinary side-effect of ending the program.  That's the sort of thing that might make it more difficult for someone to read this code later.

I feel like that could be a bit cleaner if it returned `NULL` and we checked that from the caller.  Not a big deal, though.

Reply:
Cleaner maybe, but we'd have to duplicate the NULL check, debug message, and exit(0) in 3 places in cockpit-session (plus, strictly, 6 more times in mock-auth-command). This feels more risky to me than handling EOF right here.